dev/tasks/gandiva/travis.linux.yml

```diff
@@ -0,0 +1,79 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+os: linux
+dist: trusty
+sudo: required
+language: cpp
+compiler: gcc
+services:
+  - docker
+
+# don't build twice
+if: tag IS blank
+
+env:
+  global:
+    - PLAT=x86_64
+    - TRAVIS_TAG={{ task.tag }}
+    - ARROW_TRAVIS_USE_TOOLCHAIN=1
+    - MATRIX_EVAL="CC=gcc-4.9 && CXX=g++-4.9"
+
+before_install:
+  - eval "${MATRIX_EVAL}"
+  - ulimit -c unlimited -S
+  - |
+      sudo bash -c "echo -e 'Acquire::Retries 10; Acquire::http::Timeout \"20\";' > /etc/apt/apt.conf.d/99-travis-retry"
+      sudo add-apt-repository -y ppa:ubuntu-toolchain-r/test
+      sudo apt-get update -qq
+
+before_script:
+  - git clone -b {{ arrow.branch }} {{ arrow.remote }} arrow
+  - git -C arrow checkout {{ arrow.head }}
+  - mkdir -p dist
+  - arrow/ci/travis_install_linux.sh
+  - arrow/ci/travis_install_clang_tools.sh
+  - export TRAVIS_BUILD_DIR=$TRAVIS_BUILD_DIR/arrow
+  - arrow/ci/travis_install_toolchain.sh
+  - cd arrow/cpp
+  - mkdir build
+  - cd build
+  - export PATH=$TRAVIS_BUILD_DIR/cpp-toolchain/bin:$PATH
+  - export BOOST_ROOT=$TRAVIS_BUILD_DIR/cpp-toolchain
+  - export PROTOBUF_HOME=$TRAVIS_BUILD_DIR/cpp-toolchain
+  - export LD_LIBRARY_PATH=$TRAVIS_BUILD_DIR/cpp-toolchain/lib:$LD_LIBRARY_PATH
+  - cmake .. -DCMAKE_BUILD_TYPE=Release -DARROW_GANDIVA=ON -DARROW_BUILD_UTILITIES=OFF
+  - make -j4
+  - ctest
+
+script:
+  - cd $TRAVIS_BUILD_DIR/java
+  - mvn clean install -P gandiva -Dgandiva.cpp.build.dir=../../cpp/build/release
+  - cp gandiva/target/*.jar ../../dist/
+  - cd $TRAVIS_BUILD_DIR/..
```

Comment:
Could You factor out this part to a scripts? This seems to be identical for both linux and osx.

https://github.com/apache/arrow/blob/3a893a3295e9ddec30e1f7567178e5226ba78b97/dev/tasks/gandiva/travis.linux.yml#L51-L66

Reply:
Thanks @kszucs 
To summarize, following are open items

1.Factor out common actions between the two yamls(linux and osx)
2.Add the artifacts to be verified.

Will address them both and ping back

Reply:
Thanks!